src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs

```diff
@@ -5,43 +5,101 @@ namespace Microsoft.Extensions.DependencyInjection;
-            masaStackConfig.SetValue(MasaStackConfigConstant.IS_DEMO, configuration.GetValue<bool>(MasaStackConfigConstant.IS_DEMO).ToString());
-            masaStackConfig.SetValue(MasaStackConfigConstant.DOMAIN_NAME, configuration.GetValue<string>(MasaStackConfigConstant.DOMAIN_NAME));
-            masaStackConfig.SetValue(MasaStackConfigConstant.NAMESPACE, configuration.GetValue<string>(MasaStackConfigConstant.NAMESPACE));
-            masaStackConfig.SetValue(MasaStackConfigConstant.TLS_NAME, configuration.GetValue<string>(MasaStackConfigConstant.TLS_NAME));
-            masaStackConfig.SetValue(MasaStackConfigConstant.CLUSTER, configuration.GetValue<string>(MasaStackConfigConstant.CLUSTER));
-            masaStackConfig.SetValue(MasaStackConfigConstant.OTLP_URL, configuration.GetValue<string>(MasaStackConfigConstant.OTLP_URL));
-            masaStackConfig.SetValue(MasaStackConfigConstant.REDIS, configuration.GetValue<string>(MasaStackConfigConstant.REDIS));
-            masaStackConfig.SetValue(MasaStackConfigConstant.CONNECTIONSTRING, configuration.GetValue<string>(MasaStackConfigConstant.CONNECTIONSTRING));
-            masaStackConfig.SetValue(MasaStackConfigConstant.MASA_SERVER, configuration.GetValue<string>(MasaStackConfigConstant.MASA_SERVER));
-            masaStackConfig.SetValue(MasaStackConfigConstant.MASA_UI, configuration.GetValue<string>(MasaStackConfigConstant.MASA_UI));
-            masaStackConfig.SetValue(MasaStackConfigConstant.ELASTIC, configuration.GetValue<string>(MasaStackConfigConstant.ELASTIC));
-            masaStackConfig.SetValue(MasaStackConfigConstant.ENVIRONMENT, configuration.GetValue<string>(MasaStackConfigConstant.ENVIRONMENT));
-            masaStackConfig.SetValue(MasaStackConfigConstant.ADMIN_PWD, configuration.GetValue<string>(MasaStackConfigConstant.ADMIN_PWD));
-            masaStackConfig.SetValue(MasaStackConfigConstant.DCC_SECRET, configuration.GetValue<string>(MasaStackConfigConstant.DCC_SECRET));
+            return new MasaStackConfig(configs, null);
         });
+
+        var masaStackConfig = services.GetMasaStackConfig();
+        var dccOptions = masaStackConfig.GetDefaultDccOptions();
+        services.AddMasaConfiguration(builder => builder.UseDcc(dccOptions));
+
+        var serviceProvider2 = services.BuildServiceProvider();
+        var configurationApiManage = serviceProvider2.GetRequiredService<IConfigurationApiManage>();
+        var configurationApiClient = serviceProvider2.GetRequiredService<IConfigurationApiClient>();
+
+        try
+        {
+            var remoteConfigs = await configurationApiClient.GetAsync<Dictionary<string, string>>(
+                masaStackConfig.Environment,
+                masaStackConfig.Cluster,
+                DEFAULT_PUBLIC_ID,
+                DEFAULT_CONFIG_NAME,
+                value =>
+                {
+                    masaStackConfig.UpdateMasaStackConfigContent(value);
+                });
```

Comment:
After the remote Dcc configuration is changed, it is inappropriate to change the obtained configuration immediately. In the same request, the obtained configuration should always be the same, even if the configuration is modified before the request ends